botocore/handlers.py

```diff
@@ -188,6 +191,9 @@ def copy_source_sse_md5(params, **kwargs):
 def _sse_md5(params, sse_member_prefix='SSECustomer'):
     if not _needs_s3_sse_customization(params, sse_member_prefix):
         return
+    if not md5_available():
```

Comment:
Rather than require each call site to check if md5 is available before creating checksums, can we just have a compat function that either calculate the md5 if possible or raises an exception if not available?

I'd rather have this logic once rather than throughout the code base.  It'll be easier to maintain.